### DISCUSSION TITLE
could not be inlined because the return instruction is not at the end of the function

### DISCUSSION THREAD

**Discussion by jwinarske** [google/filament#4489] [Q&A] [answered]
I have a large amount of these warnings in my build log. 

How do I run merge-return before inlining?

```WARNING: :0:0:0: The function '%92 = OpFunction %14 None %35' could not be inlined because the return instruction is not at the end of the function. This could be fixed by running merge-return before inlining.```

Thanks,
Joel

**Comment by romainguy** [comment, marked as answer]
It's a known problem with the shader optimizer you can safely ignore. You didn't do anything wrong

**Reply by jwinarske** [reply]
cool thanks!
